### ISSUE TITLE
AppSync Operations not getting unblocked after signing in again (federated)

### ISSUE THREAD

**Issue by liviu-timar** [aws-amplify/aws-sdk-android#2335]
Hello,

It's been several days since I'm trying to figure out why this is not working, so I believe it's a bug.

**Describe the bug**
Queries/mutations get blocked when federated tokens are invalid, which is normal. The UserStateListener.onUserStateChanged method gets called, I am checking if the state is SIGNED_OUT_FEDERATED_TOKENS_INVALID. If it is, I get a new token (from Firebase) and call again mobileClient.federatedSignIn. The problem is that the blocked queries are **still blocked and all future queries are blocked as well**. The funny thing is that on iOS queries get unblocked, as expected.

The docs say: If the Refresh tokens have expired and you then make call to any AWS service, such as a AppSync GraphQL request or S3 upload, the AWSMobileClient will dispatch a state notification that a re-login is required. At this point you can choose to present the user with a login screen, **call AWSMobileClient.default().signIn()**, or perform custom business logic. -> **I am doing exactly that**... Maybe I'm missing something here.

**To Reproduce**
```kotlin
//...
// Called when user state = SIGNED_OUT_FEDERATED_TOKENS_INVALID:
firebaseAuth.currentUser?.getIdToken(false)?.addOnSuccessListener { result ->
    signIn(providerUrl, result.token!!)
}
//...
fun signIn(providerUrl: String, idToken: String) {
    mobileClient.federatedSignIn(providerUrl, idToken, object : Callback<UserStateDetails> {
        override fun onResult(result: UserStateDetails?) {
            Log.d(TAG, "signIn onResponse")
        }

        override fun onError(e: Exception?) {
            Log.d(TAG, "signIn onError")
        }
    })
}
```

**Which AWS service(s) are affected?**
Mobile Client, AppSync

**Expected behavior**
Queries/mutations to be unblocked automatically, as the documentation states.

**Environment Information (please complete the following information):**
 - Mobile Client: 2.22.0
 - AppSync: 3.1.2
 - Auth Userpools: 2.21.0
 - Device: Pixel 4
 - Android Version: Android 10
 - Specific to simulators: No

Thanks for the help!

**Comment by liviu-timar** [comment]
Any help would be appreciated!

**Comment by eeatonaws** [comment]
Thank you for reporting this issue. Are you still experiencing the issue with the latest versions of the AWS SDK for Android and AWS AppSync SDK for Android?

**Comment by eeatonaws** [comment]
Closing due to inactivity. Please create a new issue or reopen this issue if you are still experiencing the issue with the latest versions of the AWS SDK for Android and AWS AppSync SDK for Android.

**Comment by liviu-timar** [comment]
Hello,

The issue is still there, why did you close the issue? With the latest
versions.

**Comment by eeatonaws** [comment]
Thank you for letting us know. I will reopen the issue.

**Comment by eeatonaws** [comment]
Hi @liviu-timar. It’s possible that `firebaseAuth.currentUser?.getIdToken(false)` is returning the old, cached ID token, rather than a new, refreshed ID token. When you observe a user state of `SIGNED_OUT_FEDERATED_TOKENS_INVALID` and then call `mobileClient.federatedSignIn(…)`, what is the user state in the onResult callback (`result.userState`)? Could you also try passing in true when you call `firebaseAuth.currentUser?.getIdToken(…)` to force refresh the token? It would also be helpful if you could provide logs showing the user state changes and logs, if there are any, about the AppSync queries being blocked.

**Comment by eeatonaws** [comment]
Closing due to inactivity. Please create a new issue or reopen this issue with the requested information if the issue persists.

**Comment by liviu-timar** [comment]
Thank you, sorry for not responding!
